Source/scrollrt.cpp

```diff
@@ -38,6 +38,131 @@
 
 namespace devilution {
 
+namespace {
+/**
+ * @brief Hash algorithm for point
+ */
+struct PointHash {
+	std::size_t operator()(Point const &s) const noexcept
+	{
+		return s.x ^ (s.y << 1);
+	}
+};
+
+/**
+ * @brief Contains all Missile at rendering position
+ */
+std::unordered_multimap<Point, MissileStruct*, PointHash> MissilesAtRenderingTile;
+
+/**
+ * @brief Could the missile (at the next game tick) collide? This method is a simplified version of CheckMissileCol (for example without random).
+ */
+bool CouldMissileCollide(Point tile, bool checkPlayerAndMonster)
+{
+	if (tile.x >= MAXDUNX || tile.x < 0)
+		return true;
+	if (tile.y >= MAXDUNY || tile.y < 0)
+		return true;
+	if (checkPlayerAndMonster) {
+		if (dMonster[tile.x][tile.y] > 0)
+			return true;
+		if (dPlayer[tile.x][tile.y] > 0)
+			return true;
+	}
+	int oid = dObject[tile.x][tile.y];
+	if (oid != 0) {
+		oid = oid > 0 ? oid - 1 : -(oid + 1);
+		if (!object[oid]._oMissFlag)
+			return true;
+	}
+	if (nMissileTable[dPiece[tile.x][tile.y]])
+		return true;
+	return false;
+}
+
+void UpdateMissileRendererData(MissileStruct &m)
+{
+	if (m.position.renderingIsFixed)
+		return;
+
+	m.position.tileForRendering = m.position.tile;
+	m.position.offsetForRendering = m.position.offset;
+
+	const MissileMovementDistrubution missileMovement = missiledata[m._mitype].MovementDistribution;
+	// don't calculate missile position if they don't move
+	if (missileMovement  == MissileMovementDistrubution::Disabled)
+		return;
+
+	// when some missiles hit, they change there animation to a explosion and the explosion shouldn't move (for example fireball)
+	if (missiledata[m._mitype].mFileNum != m._miAnimType)
+		return;
+
+	float fProgress = gfProgressToNextGameTick;
+	Displacement velocity = m.position.velocity * fProgress;
+	Displacement traveled = m.position.traveled + velocity;
+
+	int mx = traveled.deltaX >> 16;
+	int my = traveled.deltaY >> 16;
+	int dx = mx + 2 * my;
+	int dy = 2 * my - mx;
+	int lx = dx / 8;
+	dx = dx / 64;
+	int ly = dy / 8;

```

Comment:
lx and ly are unused, intentional?

Reply:
No copy&past error.
The calculation logic was copied from `UpdateMissilePos`.
And there also the offset for lightning was calculated.

so fixed it. thanks 🙂